trestle/cli.py

```diff
@@ -63,6 +66,20 @@ class Trestle(CommandPlusDocs):
         VersionCmd
     ]
 
+    discovered_plugins = {
+        name: importlib.import_module(name)
+        for finder,
+        name,
+        ispkg in pkgutil.iter_modules()
+        if name.startswith('trestle_')
+    }
+
+    logger.info(discovered_plugins)
+    if 'trestle_fedramp' in discovered_plugins:
+        fedramp_validate = importlib.import_module('trestle_fedramp.commands.validate')
+        subcommands.append(fedramp_validate.ValidateCmd)
+        logger.info('fedramp-validate command added')
+
```

Comment:
So this point here needs to be inferred, or trestle becomes a centralized choke point for plug-in capability.

I think what we want to do is to query for modules under:
```python

f'{discovered_plugins}.commands.*'
```
and then discover all classes inheriting from `ilcli.Command` (or `CommandPlusDocs` as it may be).

The method below (from generators_test.py) performs a similar task for oscal models:

```python
def test_get_all_sample_models() -> None:
    """Test we can get all models which exist."""
    pkgpath = os.path.dirname(oscal.__file__)
    for _, name, _ in pkgutil.iter_modules([pkgpath]):
        __import__(f'trestle.oscal.{name}')
        clsmembers = inspect.getmembers(sys.modules[f'trestle.oscal.{name}'], inspect.isclass)
        for _, oscal_cls in clsmembers:

            # This removes some enums and other objects.
            # add check that it is not OscalBaseModel
            if issubclass(oscal_cls, OscalBaseModel):
                gens.generate_sample_model(oscal_cls)
```

To fix this - we will need to provide a `plugins` documentation page to ensure how to interlock with this point is clear to 3rd parties.

Reply:
updated code as requested